netCDF4/_netCDF4.pyx

```diff
@@ -3754,27 +3757,64 @@ rename a `netCDF4.Variable` attribute named `oldname` to `newname`."""
             # note: missing_value has to have same endian-ness as variable
             # or this won't work.
             mval = numpy.array(self.missing_value, self.dtype)
+            # create mask from missing values. 
+            mvalmask = numpy.zeros(data.shape, numpy.bool)
+            # set mask=True for data outside valid_min,valid_max.
+            # (issue #576)
+            validmin = None; validmax = None
+            # if valid_range exists use that, otherwise
+            # look for valid_min, valid_max.  No special
+            # treatment of byte data as described at
+            # http://www.unidata.ucar.edu/software/netcdf/docs/attribute_conventions.html).
+            if hasattr(self, 'valid_range') and len(self.valid_range) == 2:
+                validmin = numpy.array(self.valid_range[0], self.dtype)
+                validmax = numpy.array(self.valid_range[1], self.dtype)
+            else:
+                if hasattr(self, 'valid_min'):
+                    validmin = numpy.array(self.valid_min, self.dtype)
+                if hasattr(self, 'valid_max'):
+                    validmax = numpy.array(self.valid_max, self.dtype)
```

Comment:
It tries to cast the attributes to the data type of the variable.  This could potentially fail, and it might be a good idea to catch this error.